Documentation/coding-guidelines/breaking-changes.md

```diff
@@ -16,47 +22,97 @@ To help triage breaking changes, we classify them in to four buckets:
+[behavioral-changes]: breaking-change-rules.md#behavioral-changes
 
-These require judgment: how predictable, obvious, consistent was the behavior?
+## Bucket 3: Unlikely Grey Area
+*Change of behavior that customers could have depended on, but probably
+wouldn't.*
+
+Examples:
 
-### Bucket 3: Unlikely Grey Area
-*Change of behavior that customers could have depended on, but probably wouldn't.*
+* Correcting behavior in a subtle corner case
 
-**Examples:**
-* correcting behavior in a subtle corner case
+As with type 2 changes, these require judgment: what is reasonable and what’s
+not?
 
-As with type 2 changes, these require judgment: what is reasonable and what’s not?
+## Bucket 4: Clearly Non-Public
+*Changes to surface area or behavior that is clearly internal or non-breaking
+in theory, but breaks an app.*
 
-### Bucket 4: Clearly Non-Public
-*Changes to surface area or behavior that is clearly internal or non-breaking in theory, but breaks an app.*
+Examples:
 
-**Examples:**
 * Changes to internal API that break private reflection
 
-It is impossible to evolve a code base without making such changes, so we don't require up-front approval for these, but we will sometimes have to go back and revisit such change if there's too much pain inflicted on the ecosystem through a popular app or library.
+It is impossible to evolve a code base without making such changes, so we don't
+require up-front approval for these, but we will sometimes have to go back and
+revisit such change if there's too much pain inflicted on the ecosystem through
+a popular app or library.
+
+This bucket is painful for the machine-wide .NET Framework, but we do have much
+more latitude here in .NET Core.
+
+## What This Means for Contributors
+
+* All bucket 1, 2, and 3 breaking changes require talking to the repo owners
+  first:
+    - We generally **don't accept** change proposals that are in bucket #1.
+    - We **might accept** change proposals that are in #2 and #3 after a
+      risk-benefit analysis. See below for more details.
+    - We **usually accept** changes that are in bucket #4
+* If you're not sure in which bucket applies to a given change, contact us as
+  well.
+
+### Risk-Benefit Analysis
+
+For buckets #2 and #3 we apply a risk-benefit analysis. It doesn't matter if the
+old behavior is "wrong", we still need to think through the implications. This
+can result in one of the following outcomes: 
+
+* **Accepted with quirking**. Depending on the estimated customer impact, we may
+  decide to "quirk" the behavior. That means, that the new behavior is only
+  exposed to apps that are compiled against the version of .NET Core that
+  introduced the change. In other words, apps running against a shared framework
+  will not see the behavior if they were compiled against an older version of
+  .NET Core and thus wouldn't have been tested in the context of the behavioral
```

Comment:
I don't really follow this section. Apps built to use a shared framework cannot run against any other version of the shared framework; they are locked to the version they built against. By definition, they must be explicitly re-built to target a different version. Therefore, I don't understand what a "quirk" means in this context. If we make a behavior-breaking change/fix in version 3.0, then existing 2.0 applications cannot possibly be affected by it, even if they are using a shared framework.

Am I misreading something here?